### PR TITLE
Add native projections to shared-mem analytics

### PR DESCRIPTION
**Pull request by gurbinder533**

### CODE REVIEW

**Comment by yan-zaretskiy** [comment]
Looks good to me. I have a question though - all these tests are marked as `NO_VERIFY`. Does it mean that we're not really checking the output results for correctness? Is the plan to eventually get rid of the `NO_VERIFY` flag?

File: lonestar/analytics/cpu/bfs/CMakeLists.txt [hunk 1]
```diff
@@ -3,3 +3,7 @@ add_dependencies(apps bfs-cpu)
 target_link_libraries(bfs-cpu PRIVATE Katana::galois lonestar)
 
 add_test_scale(small1 bfs-cpu INPUT rmat15 INPUT_URI "${RDG_RMAT15}" --edgePropertyName=value NO_VERIFY)
+
+## Test TranformView
+add_test_scale(small bfs-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person) 
+
```
Comment (uditagar) [inline lonestar/analytics/cpu/bfs/CMakeLists.txt]: can add one test for edge type as well. thanks

File: lonestar/analytics/cpu/cdlp/CMakeLists.txt [hunk 2]
```diff
@@ -4,3 +4,7 @@ target_link_libraries(cdlp-cpu PRIVATE Katana::galois lonestar)
 
 add_test_scale(small cdlp-cpu NO_VERIFY INPUT rmat15 INPUT_URI "${RDG_RMAT15}" --algo=Synchronous)
 add_test_scale(small cdlp-cpu NO_VERIFY INPUT rmat15 INPUT_URI "${RDG_RMAT15_SYMMETRIC}" -symmetricGraph --algo=Synchronous)
+
+## Test TranformView
+add_test_scale(small cdlp-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person) 
+
```
Comment (uditagar) [inline lonestar/analytics/cpu/cdlp/CMakeLists.txt]: need an edge type test

File: lonestar/analytics/cpu/connected-components/CMakeLists.txt [hunk 3]
```diff
@@ -13,3 +13,6 @@ add_test_scale(small connected-components-cpu NO_VERIFY INPUT rmat15 INPUT_URI "
 add_test_scale(small connected-components-cpu NO_VERIFY INPUT rmat15 INPUT_URI "${RDG_RMAT15}" "-algo=Asynchronous")
 add_test_scale(small connected-components-cpu NO_VERIFY INPUT rmat15 INPUT_URI "${RDG_RMAT15}" "-algo=Afforest")
 add_test_scale(small connected-components-cpu NO_VERIFY INPUT rmat15 INPUT_URI "${RDG_RMAT15}" "-algo=EdgeTiledAfforest")
+
+## Test TranformView
+add_test_scale(small connected_components-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person -algo=EdgeTiledAfforest)
```
Comment (uditagar) [inline lonestar/analytics/cpu/connected-components/CMakeLists.txt]: add edge type test

File: lonestar/analytics/cpu/k-core/CMakeLists.txt [hunk 4]
```diff
@@ -7,3 +7,6 @@ add_test_scale(small k-core-cpu INPUT rmat15 INPUT_URI "${RDG_RMAT15_SYMMETRIC}"
 
 add_test_scale(small k-core-cpu INPUT rmat15 INPUT_URI "${RDG_RMAT15}" --kCoreNumber=100 --algo=Synchronous)
 add_test_scale(small k-core-cpu INPUT rmat15 INPUT_URI "${RDG_RMAT15}" --kCoreNumber=100 --algo=Asynchronous)
+
+## Test TranformView
+add_test_scale(small k-core-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person)
```
Comment (uditagar) [inline lonestar/analytics/cpu/k-core/CMakeLists.txt:12]: add edge type test

File: lonestar/analytics/cpu/k-shortest-paths/CMakeLists.txt [hunk 5]
```diff
@@ -3,3 +3,6 @@ add_dependencies(apps k-shortest-paths-cpu)
 target_link_libraries(k-shortest-paths-cpu PRIVATE Katana::graph lonestar)
 
 add_test_scale(small1 k-shortest-paths-cpu NO_VERIFY INPUT rmat15 INPUT_URI "${RDG_RMAT15}" -delta=8 --edgePropertyName=value)
+
+## Test TranformView
+add_test_scale(small k-shortest-paths-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person)
```
Comment (uditagar) [inline lonestar/analytics/cpu/k-shortest-paths/CMakeLists.txt:25]: add edge type test

File: lonestar/analytics/cpu/leiden_clustering/CMakeLists.txt [hunk 6]
```diff
@@ -7,3 +7,7 @@ add_test_scale(small leiden-clustering-cpu NO_VERIFY INPUT rmat10 INPUT_URI "${R
 
 add_test_scale(small leiden-clustering-cpu NO_VERIFY INPUT rmat10 INPUT_URI "${RDG_RMAT10}" --edgePropertyName=value --algo=DoAll)
 add_test_scale(small leiden-clustering-cpu NO_VERIFY INPUT rmat10 INPUT_URI "${RDG_RMAT10}" --edgePropertyName=value --algo=Deterministic)
+
+## Test TranformView
+add_test_scale(small leiden-clustering-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person --algo=Deterministic)
```
Comment (uditagar) [inline lonestar/analytics/cpu/leiden_clustering/CMakeLists.txt:12]: add edge type test

File: lonestar/analytics/cpu/louvain_clustering/CMakeLists.txt [hunk 7]
```diff
@@ -8,6 +8,5 @@ add_test_scale(small louvain-clustering-cpu NO_VERIFY INPUT rmat10 INPUT_URI "${
 add_test_scale(small louvain-clustering-cpu NO_VERIFY INPUT rmat10 INPUT_URI "${RDG_RMAT10}" --edgePropertyName=value --algo=Deterministic)
 add_test_scale(small louvain-clustering-cpu NO_VERIFY INPUT rmat10 INPUT_URI "${RDG_RMAT10}" --edgePropertyName=value --algo=DoAll)
 
-
-
-
+## Test TranformView
+add_test_scale(small louvain-clustering-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person --algo=Deterministic)
```
Comment (uditagar) [inline lonestar/analytics/cpu/louvain_clustering/CMakeLists.txt:12]: add edge type test

File: lonestar/analytics/cpu/pagerank/CMakeLists.txt [hunk 8]
```diff
@@ -18,3 +18,5 @@ add_test_scale(small pagerank-cpu
   INPUT rmat15 INPUT_URI "${RDG_RMAT15}" REL_TOL 0.01 MEAN_TOL 0.002
   -maxIterations=100 -algo=PullResidual)
 
+## Test TranformView
+add_test_scale(small pagerank-cpu NO_VERIFY INPUT ldbc003 INPUT_URI "${RDG_LDBC_003}" --node_types=Person)
```
Comment (uditagar) [inline lonestar/analytics/cpu/pagerank/CMakeLists.txt]: add edge type test

**Review by uditagar — APPROVED** [review]
looks fine

**Review by uditagar — APPROVED** [review]
looks fine now
